migrations/20200927132044-update-patient-barcode-non-nullable.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface) => {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.sequelize.query(
+          `UPDATE patients SET "barcodeValue" = "id" WHERE "barcodeValue" IS NULL`,
+          { transaction: t }
+        ),
+        queryInterface.sequelize.query(
+          'ALTER TABLE patients ALTER COLUMN "barcodeValue" TYPE "varchar", ALTER COLUMN "barcodeValue" SET NOT NULL',
+          { transaction: t }
+        ),
+      ]);
+    });
+  },
+
+  down: (queryInterface) => {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.sequelize.query(
+          'ALTER TABLE patients ALTER COLUMN "barcodeValue" TYPE "varchar", ALTER COLUMN "barcodeValue" DROP NOT NULL',
+          { transaction: t }
+        ),
+        queryInterface.sequelize.query(
+          `UPDATE patients SET "barcodeValue" = NULL WHERE "barcodeValue" = "id"::varchar`,
+          { transaction: t }
+        ),
+      ]);
+    });
```

Comment:
`Promise.all` runs everything in its array concurrently - this means that order is not guaranteed

to ensure that the order is followed, we can use `await`/`async` syntax and do something like this (please test as i just threw this together quickly) - this can also be done in the `up` mutation for the same purpose
```suggestion
    return queryInterface.sequelize.transaction(async (t) => {
      await queryInterface.sequelize.query(
          'ALTER TABLE patients ALTER COLUMN "barcodeValue" TYPE "varchar", ALTER COLUMN "barcodeValue" DROP NOT NULL',
          { transaction: t }
        );
      return queryInterface.sequelize.query(
          `UPDATE patients SET "barcodeValue" = NULL WHERE "barcodeValue" = "id"::varchar`,
          { transaction: t }
        );
    });
```